hyperspy/axes.py

```diff
@@ -688,46 +805,18 @@ def value_range_to_indices(self, v1, v2):
                 i2 = self.value2index(v2)
         return i1, i2
 
-    def update_from(self, axis, attributes):
-        """Copy values of specified axes fields from the passed AxesManager.
-
-        Parameters
-        ----------
-        axis : BaseDataAxis
-            The BaseDataAxis instance to use as a source for values.
-        attributes : iterable container of strings.
-            The name of the attribute to update. If the attribute does not
-            exist in either of the AxesManagers, an AttributeError will be
-            raised.
-
-        Returns
-        -------
-        A boolean indicating whether any changes were made.
-
-        """
-        any_changes = False
-        changed = {}
-        for f in attributes:
-            if getattr(self, f) != getattr(axis, f):
-                changed[f] = getattr(axis, f)
-        if len(changed) > 0:
-            self.trait_set(**changed)
-            any_changes = True
-        return any_changes
-
     def convert_to_uniform_axis(self):
         """Convert to an uniform axis."""
         scale = (self.high_value - self.low_value) / self.size
+        offset = self.low_value
+        size = self.size
         d = self.get_axis_dictionary()
-        axes_manager = self.axes_manager
-        del d["axis"]
         if len(self.axis) > 1:
             scale_err = max(self.axis[1:] - self.axis[:-1]) - scale
             _logger.warning('The maximum scale error is {}.'.format(scale_err))
         d["_type"] = 'UniformDataAxis'
-        self.__class__ = UniformDataAxis
-        self.__init__(**d, size=self.size, scale=scale, offset=self.low_value)
-        self.axes_manager = axes_manager
+        new_axis = UniformDataAxis(**d, size=size, scale=scale, offset=offset)
```

Comment:
It is convenient to convert inplace and it would be good to avoid returning a different object.

Reply:
I can see if there is a way to do it. I think it means that we have manage how the changes to different attributes are handled which is a bit more involved.

Reply:
I think I'll become an expert at the Traits package by the end of this :)

Reply:
@ericpre Well it only took me a couple of days but how do you feel about this.


```python

    def convert_to_non_uniform_axis(self):
        """Convert to a non-uniform axis."""
        d = super().get_axis_dictionary()
        d["_type"] = 'DataAxis'
        new_axis = DataAxis(**d)
        if self.axes_manager is not None:
            self.axes_manager[self] = new_axis
        else:
            return new_axis
```

Reply:
The problem with returning a new object is that the events connected to the old one will be lost.